SConstruct

```diff
@@ -80,11 +81,15 @@ class FreelanEnvironment(Environment):
             self.install_prefix = self.prefix
             self.bin_install_prefix = self.bin_prefix
 
-        if os.path.basename(self['CXX']) == 'clang++':
-            self.Append(CXXFLAGS=['-Qunused-arguments'])
-            self.Append(CXXFLAGS=['-fcolor-diagnostics'])
-        elif os.path.basename(self['CXX']).startswith('g++'):
-            self.Append(CXXFLAGS=['-Wno-missing-field-initializers'])
+        if self['CXX'] is None:
```

Comment:
Ah...! But we have a problem : **SCons seems to disable assertions at compile time.** (I've tested it)

Just as warned [here](https://wiki.python.org/moin/UsingAssertionsEffectively) :

> Assertions should *not* be used to test for failure cases that can occur because of bad user input or operating system/environment failures, such as a file not being found. Instead, you should raise an exception, or print an error message, or whatever is appropriate. One important reason why assertions should only be used for self-tests of the program is that assertions can be disabled at compile time.

> If Python is started with the -O option, then assertions will be stripped out and not evaluated.